src/canvas-view/canvas/canvas-component.js

```diff
@@ -0,0 +1,406 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+    .filter('trustAsResourceUrl', ['$sce', function ($sce) {
+      return function (val) {
+        return $sce.trustAsResourceUrl(val);
+      };
+    }])
+
+    //
+    // Component that generates the rendered chart from the data model.
+    //
+    .component('pfCanvas', {
+      templateUrl: "canvas-view/canvas/canvas.html",
+      bindings: {
+        chartDataModel: "=",
+        chartViewModel: "=?",
+        readOnly: "=?",
+        hideConnectors: "=?"
+      },
+      controller: function CanvasController ($scope, dragging, $element, $document) {
+        var ctrl = this;
+
+        ctrl.chart = new pfCanvas.ChartViewModel(ctrl.chartDataModel);
+        ctrl.chartViewModel = ctrl.chart;
+
+        //
+        // Init data-model variables.
+        //
+        ctrl.draggingConnection = false;
+        ctrl.connectorSize = 6;
+        ctrl.dragSelecting = false;
+
+        //
+        // Reference to the connection, connector or node that the mouse is currently over.
+        //
+        ctrl.mouseOverConnector = null;
+        ctrl.mouseOverConnection = null;
+        ctrl.mouseOverNode = null;
+
+
+        //
+        // Translate the coordinates so they are relative to the svg element.
+        //
+        this.translateCoordinates = function (x, y, evt) {
+          var svgElem =  $element.get(0).children[0];
+          var matrix = svgElem.getScreenCTM();
+          var point = svgElem.createSVGPoint();
+          point.x = (x - evt.view.pageXOffset) / ctrl.zoomLevel();
+          point.y = (y - evt.view.pageYOffset) / ctrl.zoomLevel();
+
+          return point.matrixTransform(matrix.inverse());
+        };
+
+        ctrl.hideConnectors = ctrl.hideConnectors ? ctrl.hideConnectors : false;
```

Comment:
You'll have to check it to confirm, but this might be a good spot for some simplification... 
```js
ctrl.hideConnectors = ctrl.hideConnectors || false;
```

Reply:
I agree

Reply:
If the idea is that any value makes this true, then I would use:
```js
ctrl.hideConnectors = !!ctrl.hideConnectors;
```

Reply:
It appears that it’s already considered a boolean, I could cast it but I think the syntax is a bit confusing